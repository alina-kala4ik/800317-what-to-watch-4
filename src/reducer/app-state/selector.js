import {NameSpace} from "./../name-space.js";

const getPlayableMovie = (state) => {
  return state[NameSpace.APP_STATE].playableMovie;
};

const getCountDisplayedFilms = (state) => {
  return state[NameSpace.APP_STATE].countDisplayedFilms;
};

const getGenre = (state) => {
  return state[NameSpace.APP_STATE].genre;
};

const gerServerStatus = (state) => {
  return state[NameSpace.APP_STATE].serverStatus;
};

const getLogIn = (state) => {
  return state[NameSpace.APP_STATE].logIn;
};

export {getPlayableMovie, getCountDisplayedFilms, getGenre, gerServerStatus, getLogIn};

