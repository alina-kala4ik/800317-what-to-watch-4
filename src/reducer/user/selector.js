import {NameSpace} from "./../name-space.js";

const getAuthorizationStatus = (state) => {
  return state[NameSpace.USER].authorizationStatus;
};

const getAvatar = (state) => {
  return state[NameSpace.USER].avatar;
};

const getFlagIsFetchingAuthStatus = (state) => {
  return state[NameSpace.USER].isFetchingAuthStatus;
};

const getFlagLoginDataValid = (state) => {
  return state[NameSpace.USER].isLoginDataValid;
}

export {getAuthorizationStatus, getAvatar, getFlagIsFetchingAuthStatus, getFlagLoginDataValid};
