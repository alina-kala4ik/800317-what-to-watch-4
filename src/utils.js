const extend = (a, b) => {
  return Object.assign({}, a, b);
};

const Genres = {
  ALL: `All genres`,
  COMEDIES: `Comedies`,
  CRIME: `Crime`,
  DOCUMENTARY: `Documentary`,
  DRAMA: `Drama`,
  HORROR: `Horror`,
  KIDS_FAMILY: `Kids & Family`,
  ROMANCE: `Romance`,
  SCI_FI: `Sci-Fi`,
  THRILLERS: `Thrillers`,
};

const Tabs = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};

const Pages = {
  ROOT: `/`,
  LOGIN: `/login`,
  MY_LIST: `/mylist`,
  FILM: `/films/:id?`,
  REVIEW: `/films/:id?/review`,
  PLAYER: `/films/:id?/player`,
};

export {extend, Genres, Tabs, Pages};
