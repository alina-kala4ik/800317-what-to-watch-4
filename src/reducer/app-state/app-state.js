import {extend, Genres} from "./../../utils.js";

const DISPLAYED_NUMBER_OF_FILMS = 8;

const ServerStatus = {
  ERROR: `ERROR`,
  OK: `OK`
};

const initialState = {
  genre: Genres.ALL,
  countDisplayedFilms: DISPLAYED_NUMBER_OF_FILMS,
  serverStatus: ServerStatus.OK,
};

const ActionTypes = {
  CHANGE_GENRE: `CHANGE_GENRE`,
  INCREASE_COUNT_DISPLAYED_FILMS: `INCREASE_COUNT_DISPLAYED_FILMS`,
  RESET_COUNT_DISPLAYED_FILMS: `RESET_COUNT_DISPLAYED_FILMS`,
  CHANGE_SERVER_STATUS_ON_ERROR: `CHANGE_SERVER_STATUS_ON_ERROR`,
};

const ActionCreator = {
  changeGenre: (genre) => ({
    type: ActionTypes.CHANGE_GENRE,
    payload: genre
  }),
  increaseCountDisplayedFilms: () => ({
    type: ActionTypes.INCREASE_COUNT_DISPLAYED_FILMS,
    payload: DISPLAYED_NUMBER_OF_FILMS
  }),
  resetCountDisplayedFilms: () => ({
    type: ActionTypes.RESET_COUNT_DISPLAYED_FILMS,
    payload: DISPLAYED_NUMBER_OF_FILMS
  }),
  changeServerStatusOnError: () => ({
    type: ActionTypes.CHANGE_SERVER_STATUS_ON_ERROR,
    payload: ServerStatus.ERROR
  })
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionTypes.CHANGE_GENRE:
      return extend(state, {
        genre: action.payload
      });
    case ActionTypes.INCREASE_COUNT_DISPLAYED_FILMS:
      return extend(state, {
        countDisplayedFilms: state.countDisplayedFilms + action.payload
      });
    case ActionTypes.RESET_COUNT_DISPLAYED_FILMS:
      return extend(state, {
        countDisplayedFilms: action.payload
      });
    case ActionTypes.CHANGE_SERVER_STATUS_ON_ERROR:
      return extend(state, {
        serverStatus: action.payload
      });
  }
  return state;
};

export {reducer, ActionTypes, ActionCreator, ServerStatus};
