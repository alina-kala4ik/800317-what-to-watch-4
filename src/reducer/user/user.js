import {extend} from "./../../utils.js";
import {ActionCreator as AppStateActionCreator} from "./../app-state/app-state.js";

const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const initialState = {
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  avatar: null,
};


const ActionTypes = {
  REQUIRED_AUTHORIZATION: `REQUIRED_AUTHORIZATION`,
  ADD_AVATAR: `ADD_AVATAR`,
};

const ActionCreator = {
  requiredAuthorization: (status)=>({
    type: ActionTypes.REQUIRED_AUTHORIZATION,
    payload: status,
  }),
  addAvatar: (url)=>({
    type: ActionTypes.ADD_AVATAR,
    payload: url
  }),
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionTypes.REQUIRED_AUTHORIZATION:
      return extend(state, {
        authorizationStatus: action.payload
      });
    case ActionTypes.ADD_AVATAR:
      return extend(state, {
        avatar: action.payload
      });
  }
  return state;
};

const Operation = {
  checkAuth: () => (dispatch, getState, api) => {
    return api.get(`/login`)
      .then(()=>{
        dispatch(ActionCreator.requiredAuthorization(AuthorizationStatus.AUTH));
      })
      .catch((err)=>{
        throw err;
      });
  },

  login: (authData) => (dispatch, getState, api) => {
    return api.post(`/login`, {
      email: authData.email,
      password: authData.password
    })
      .then((response)=>{
        const {avatar_url: avatarUrl} = response.data;

        dispatch(ActionCreator.requiredAuthorization(AuthorizationStatus.AUTH));
        dispatch(ActionCreator.addAvatar(avatarUrl));
        dispatch(AppStateActionCreator.logIn(false));
      })
      .catch((err)=>{
        throw err;
      });
  }
};

export {ActionCreator, reducer, Operation, AuthorizationStatus, ActionTypes};
