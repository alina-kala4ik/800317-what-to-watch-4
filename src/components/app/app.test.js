import React from "react";
import renderer from "react-test-renderer";
import App from "./app.jsx";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import {Genres} from "./../../utils.js";
import {NameSpace} from "./../../reducer/name-space.js";
import {ServerStatus} from "./../../reducer/app-state/app-state.js";
import {AuthorizationStatus} from "./../../reducer/user/user.js";

const films = [
  {
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    screenshotSrc: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    posterSrc: `img/bg-the-grand-budapest-hotel.jpg`,
    movieCoverSrc: `img/the-grand-budapest-hotel-poster.jpg`,
    genre: `Drama`,
    yearRelease: 2017,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.`,
    rating: 5.6,
    numberVotes: 278,
    producer: `Wes Andreson`,
    actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    videoSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    runTime: 99,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    isFavorite: false,
    id: 1,
  },
  {
    title: `Bohemian Rhapsody`,
    screenshotSrc: `img/bohemian-rhapsody.jpg`,
    posterSrc: `img/bg-the-grand-budapest-hotel.jpg`,
    movieCoverSrc: `img/the-grand-budapest-hotel-poster.jpg`,
    genre: `Drama`,
    yearRelease: 2017,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.`,
    rating: 5.6,
    numberVotes: 278,
    producer: `Wes Andreson`,
    actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    videoSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    runTime: 99,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    isFavorite: false,
    id: 2,
  },
  {
    title: `Macbeth`,
    screenshotSrc: `img/macbeth.jpg`,
    posterSrc: `img/bg-the-grand-budapest-hotel.jpg`,
    movieCoverSrc: `img/the-grand-budapest-hotel-poster.jpg`,
    genre: `Drama`,
    yearRelease: 2017,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.`,
    rating: 5.6,
    numberVotes: 278,
    producer: `Wes Andreson`,
    actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    videoSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    runTime: 99,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    isFavorite: false,
    id: 3,
  },
  {
    title: `Aviator`,
    screenshotSrc: `img/aviator.jpg`,
    posterSrc: `img/bg-the-grand-budapest-hotel.jpg`,
    movieCoverSrc: `img/the-grand-budapest-hotel-poster.jpg`,
    genre: `Drama`,
    yearRelease: 2017,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.`,
    rating: 5.6,
    numberVotes: 278,
    producer: `Wes Andreson`,
    actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    videoSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    runTime: 99,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    isFavorite: false,
    id: 4,
  },
  {
    title: `We need to talk about Kevin`,
    screenshotSrc: `img/we-need-to-talk-about-kevin.jpg`,
    posterSrc: `img/bg-the-grand-budapest-hotel.jpg`,
    movieCoverSrc: `img/the-grand-budapest-hotel-poster.jpg`,
    genre: `Drama`,
    yearRelease: 2017,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.`,
    rating: 5.6,
    numberVotes: 278,
    producer: `Wes Andreson`,
    actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    videoSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    runTime: 99,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    isFavorite: false,
    id: 5,
  },
  {
    title: `What We Do in the Shadows`,
    screenshotSrc: `img/what-we-do-in-the-shadows.jpg`,
    posterSrc: `img/bg-the-grand-budapest-hotel.jpg`,
    movieCoverSrc: `img/the-grand-budapest-hotel-poster.jpg`,
    genre: `Drama`,
    yearRelease: 2017,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.`,
    rating: 5.6,
    numberVotes: 278,
    producer: `Wes Andreson`,
    actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    videoSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    runTime: 99,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    isFavorite: false,
    id: 6,
  },
  {
    title: `Revenant`,
    screenshotSrc: `img/revenant.jpg`,
    posterSrc: `img/bg-the-grand-budapest-hotel.jpg`,
    movieCoverSrc: `img/the-grand-budapest-hotel-poster.jpg`,
    genre: `Drama`,
    yearRelease: 2017,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.`,
    rating: 5.6,
    numberVotes: 278,
    producer: `Wes Andreson`,
    actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    videoSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    runTime: 99,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    isFavorite: false,
    id: 7,
  },
  {
    title: `Johnny English`,
    screenshotSrc: `img/johnny-english.jpg`,
    posterSrc: `img/bg-the-grand-budapest-hotel.jpg`,
    movieCoverSrc: `img/the-grand-budapest-hotel-poster.jpg`,
    genre: `Drama`,
    yearRelease: 2017,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.`,
    rating: 5.6,
    numberVotes: 278,
    producer: `Wes Andreson`,
    actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    videoSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    runTime: 99,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    isFavorite: false,
    id: 8,
  },
];

const mockStore = configureStore([]);

it(`render App without server error`, () => {
  const store = mockStore({
    [NameSpace.APP_STATE]: {
      genre: Genres.ALL,
      countDisplayedFilms: 8,
      serverStatus: ServerStatus.OK,
    },
    [NameSpace.DATA]: {
      films,
      promoFilm: films[0],
      allFilms: films,
      isFilmsFetching: false,
      isPromoFilmFetching: false,
      genreForFilter: Genres.ALL
    },
    [NameSpace.USER]: {
      authorizationStatus: AuthorizationStatus.AUTH,
      avatar: `img/avatar.jpg`
    }
  });

  const tree = renderer
    .create(
        <Provider store={store}>
          <App
            serverStatus={ServerStatus.OK}
            isFilmsFetching={false}
            isPromoFilmFetching={false}
            films={films}
            promoFilm={films[0]}
          />
        </Provider>, {
          createNodeMock: ()=>{
            return {};
          }
        })
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it(`render App with server error`, () => {
  const store = mockStore({
    [NameSpace.APP_STATE]: {
      genre: Genres.ALL,
      countDisplayedFilms: 8,
      serverStatus: ServerStatus.ERROR,
    },
    [NameSpace.DATA]: {
      films,
      promoFilm: films[0],
      allFilms: films,
      isFilmsFetching: false,
      isPromoFilmFetching: false,
      genreForFilter: Genres.ALL
    },
    [NameSpace.USER]: {
      authorizationStatus: AuthorizationStatus.AUTH,
      avatar: `img/avatar.jpg`
    }
  });

  const tree = renderer
    .create(
        <Provider store={store}>
          <App
            serverStatus={ServerStatus.ERROR}
            isFilmsFetching={false}
            isPromoFilmFetching={false}
            films={films}
            promoFilm={films[0]}
          />
        </Provider>, {
          createNodeMock: ()=>{
            return {};
          }
        })
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it(`render App when films null`, () => {
  const store = mockStore({
    [NameSpace.APP_STATE]: {
      genre: Genres.ALL,
      countDisplayedFilms: 8,
      serverStatus: ServerStatus.OK,
    },
    [NameSpace.DATA]: {
      films: null,
      promoFilm: films[0],
      allFilms: films,
      isFilmsFetching: false,
      isPromoFilmFetching: false,
      genreForFilter: Genres.ALL
    },
    [NameSpace.USER]: {
      authorizationStatus: AuthorizationStatus.AUTH,
      avatar: `img/avatar.jpg`
    }
  });

  const tree = renderer
    .create(
        <Provider store={store}>
          <App
            serverStatus={ServerStatus.OK}
            isFilmsFetching={false}
            isPromoFilmFetching={false}
            films={null}
            promoFilm={films[0]}
          />
        </Provider>, {
          createNodeMock: ()=>{
            return {};
          }
        })
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it(`render App when promoFilm null`, () => {
  const store = mockStore({
    [NameSpace.APP_STATE]: {
      genre: Genres.ALL,
      countDisplayedFilms: 8,
      serverStatus: ServerStatus.OK,
    },
    [NameSpace.DATA]: {
      films,
      promoFilm: null,
      allFilms: films,
      isFilmsFetching: false,
      isPromoFilmFetching: false,
      genreForFilter: Genres.ALL
    },
    [NameSpace.USER]: {
      authorizationStatus: AuthorizationStatus.AUTH,
      avatar: `img/avatar.jpg`
    }
  });

  const tree = renderer
    .create(
        <Provider store={store}>
          <App
            serverStatus={ServerStatus.OK}
            isFilmsFetching={false}
            isPromoFilmFetching={false}
            films={films}
            promoFilm={null}
          />
        </Provider>, {
          createNodeMock: ()=>{
            return {};
          }
        })
    .toJSON();
  expect(tree).toMatchSnapshot();
});
