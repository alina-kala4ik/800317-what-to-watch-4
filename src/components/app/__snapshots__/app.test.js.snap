// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render App when films null 1`] = `
<div
  style={
    Object {
      "backgroundColor": "red",
    }
  }
>
  У нашего сервера лапки
</div>
`;

exports[`render App when promoFilm null 1`] = `
<div
  style={
    Object {
      "backgroundColor": "red",
    }
  }
>
  У нашего сервера лапки
</div>
`;

exports[`render App with server error 1`] = `
<div
  style={
    Object {
      "backgroundColor": "red",
    }
  }
>
  Сервер не доступен
</div>
`;

exports[`render App without server error 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="Fantastic Beasts: The Crimes of Grindelwald"
        src="img/bg-the-grand-budapest-hotel.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <a
          className="user-block__avatar"
          href="/mylist"
          onClick={[Function]}
        >
          <img
            alt="User avatar"
            height="63"
            src="img/avatar.jpg"
            width="63"
          />
        </a>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="Fantastic Beasts: The Crimes of Grindelwald poster"
            height="327"
            src="img/the-grand-budapest-hotel-poster.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Fantastic Beasts: The Crimes of Grindelwald
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Drama
            </span>
            <span
              className="movie-card__year"
            >
              2017
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item "
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Drama
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__image"
            href="/films/1"
            onClick={[Function]}
          >
            <video
              autoPlay={false}
              height="175"
              loop={true}
              poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
              preload="none"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width="280"
            />
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/1"
              onClick={[Function]}
            >
              Fantastic Beasts: The Crimes of Grindelwald
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__image"
            href="/films/2"
            onClick={[Function]}
          >
            <video
              autoPlay={false}
              height="175"
              loop={true}
              poster="img/bohemian-rhapsody.jpg"
              preload="none"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width="280"
            />
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/2"
              onClick={[Function]}
            >
              Bohemian Rhapsody
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__image"
            href="/films/3"
            onClick={[Function]}
          >
            <video
              autoPlay={false}
              height="175"
              loop={true}
              poster="img/macbeth.jpg"
              preload="none"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width="280"
            />
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/3"
              onClick={[Function]}
            >
              Macbeth
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__image"
            href="/films/4"
            onClick={[Function]}
          >
            <video
              autoPlay={false}
              height="175"
              loop={true}
              poster="img/aviator.jpg"
              preload="none"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width="280"
            />
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/4"
              onClick={[Function]}
            >
              Aviator
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__image"
            href="/films/5"
            onClick={[Function]}
          >
            <video
              autoPlay={false}
              height="175"
              loop={true}
              poster="img/we-need-to-talk-about-kevin.jpg"
              preload="none"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width="280"
            />
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/5"
              onClick={[Function]}
            >
              We need to talk about Kevin
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__image"
            href="/films/6"
            onClick={[Function]}
          >
            <video
              autoPlay={false}
              height="175"
              loop={true}
              poster="img/what-we-do-in-the-shadows.jpg"
              preload="none"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width="280"
            />
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/6"
              onClick={[Function]}
            >
              What We Do in the Shadows
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__image"
            href="/films/7"
            onClick={[Function]}
          >
            <video
              autoPlay={false}
              height="175"
              loop={true}
              poster="img/revenant.jpg"
              preload="none"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width="280"
            />
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/7"
              onClick={[Function]}
            >
              Revenant
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__image"
            href="/films/8"
            onClick={[Function]}
          >
            <video
              autoPlay={false}
              height="175"
              loop={true}
              poster="img/johnny-english.jpg"
              preload="none"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width="280"
            />
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/8"
              onClick={[Function]}
            >
              Johnny English
            </a>
          </h3>
        </article>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          © 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
